src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      sortTableByColumn(index);
+    });
+  });
+
+  function sortTableByColumn(columnIndex) {
+    const sortedRows = rows.slice().sort((rowA, rowB) => {
+      const cellA = rowA.cells[columnIndex].textContent.trim();
+      const cellB = rowB.cells[columnIndex].textContent.trim();
+
+      if (columnIndex === 2) {
+        return parseInt(cellA) - parseInt(cellB);
+      }
+
+      if (columnIndex === 3) {
+        return (
+          parseFloat(cellA.replace(/[$,]/g, '')) -
+          parseFloat(cellB.replace(/[$,]/g, ''))
+        );
+      }
+
+      return cellA.localeCompare(cellB);
```

Comment:
The default sorting logic uses `localeCompare`, which is suitable for strings. Ensure that this is the desired behavior for columns not explicitly handled (i.e., columns other than 2 and 3). If other data types are present, consider adding additional conditions to handle them appropriately.